gw_eccentricity/eccDefinition.py

```diff
@@ -174,6 +176,11 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
                 omega_gw_apocenters(t).
                 Defaults are set using utils.get_default_spline_kwargs
 
+            rational_fit_kwargs: dict
+                Dictionary of arguments to be passed to the rational
+                fit function. Defaults are set using
+                `utils.get_default_rational_fit_kwargs`
+
```

Comment:
You should indicate here already that only one of spline_kwargs or rational_fit_kwargs are allowed.
Question: Would it be better to have interp_kwargs, which has a key 'interp_type' that can be 'spline' or 'rational' fit? In the future, if we add extra options like Toni's PN inspired fits, that would make it easier.
This is more of a question for you rather than a suggestion because I don't know if that will make the code more complicated than it needs to be. Either way, needs to be clear that only one is allowed.

Reply:
We use spline kwargs for things other than interpolating the omega_gw extrema. For example, we interpolate zeroecc data using spline. So providing just rational_fit kwargs won't work because rational_fit is only good for interpolating monotonic trends.

Reply:
Basically we will need to make all interpolation thing work with a single method if we want the user to provide only one of them. Here, we wanted to use rational fit for the omega envelope mainly.

Reply:
What do you suggest?